ModelVale/ModelVale/ViewControllers/Main/ModelViewController.m

```diff
@@ -54,4 +100,160 @@ - (IBAction)didTapLogout:(id)sender {
     }];
 }
 
+//MARK: XP animations
+
+- (CAShapeLayer*) getXPBubbleLayer {
+    CAShapeLayer* XPLayer = [CAShapeLayer new];
+    XPLayer.fillColor = [UIColor clearColor].CGColor;
+    return XPLayer;
+}
+
+- (UIBezierPath*) getXPLoopPath: (CGPoint)XPStart XPEnd: (CGPoint) XPEnd {
+    UIBezierPath* path = [UIBezierPath new];
+    [path moveToPoint:XPStart];
+    CGPoint controlOne = CGPointMake(XPEnd.x - 0.5*(XPStart.x - XPEnd.x), XPStart.y - 50);
```

Comment:
some more magic numbers here ie - 0.5 and 50

Reply:
Honestly the whole  path is sort of a magic number -- it's just any path that looks cool and ends at XPEnd. There isn't much structure to it which is why I just left the magic numbers. I could make them constants, but the formulas to get the control points are also somewhat arbitrary. What do you think?